### PR TITLE
Update documentation for std::process::Command's new method

### PR DESCRIPTION
**Pull request by sanchopanca**
In the current documentation, it's not specified that when creating a Command, the .exe extension can be omitted for Windows executables. However, for other types of executable files like .bat or .cmd, the complete filename including the extension must be provided.

I encountered it by noticing that `Command::new("wt").spawn().unwrap()` succeeds on my machine while `Command::new("code").spawn().unwrap()` panics. Turns out VS Code's entrypoint is .cmd file.

`resolve_exe` method mentions this behaviour in [a comment](https://github.com/rust-lang/rust/blob/e7fda447e7d05b6ca431fc8fe8f489b1fda810bc/library/std/src/sys/windows/process.rs#L425), but it makes sense to mention it at a more visible place.

I've added this clarification to the documentation, which should make it more accurate and helpful for Rust developers working on the Windows platform.

### CODE REVIEW

**Comment by ChrisDenton** [comment]
I would love better documentation here but I'm also a bit reluctant to encourage people to pass `.bat` files directly to `Command::new` instead of doing `Command::new('cmd.exe').args(&["/c", "script.bat"])` as the former is not well supported (though it should hopefully work in that simple case).

Basically we use `CreateProcessW` under the hood [which says](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw#parameters):

> To run a batch file, you must start the command interpreter; set `lpApplicationName` to `cmd.exe` and set `lpCommandLine` to the following arguments: `/c` plus the name of the batch file.

Basically it's only for running `.exe` files and not script files. But it turned out this wasn't quite true. An undocumented feature is that it did attempt to do this transformation for you when given a `.bat` file (or `.cmd`). So when, for security reasons, Rust switched to using its own code to resolve exe paths we attempted to emulate this so as to avoid breaking people's existing code. However, I would still be nervous about people relying on that.

So, yes, I'm in favour of documenting the behaviour but I'm less certain about referencing "other executable files" because as far as `CreateProcessW` is concerned there's officially only one type of executable file.

**Comment by cuviper** [comment]
If you don't mind, I'll let you own this review. :)

r? @ChrisDenton

File: library/std/src/process.rs [hunk 1]
```diff
@@ -558,6 +558,12 @@ impl Command {
     /// but this has some implementation limitations on Windows
     /// (see issue #37519).
     ///
+    /// Note on Windows: For executable files with the .exe extension,
+    /// it can be omitted when specifying the program for this Command.
+    /// However, for other types of executable files like .bat or .cmd,
+    /// a filename including the extension needs to be provided,
+    /// otherwise the file won't be found.
```
Comment (ChrisDenton) [inline library/std/src/process.rs:565]: How about something like:
```suggestion
    /// However, if the file has a different extension
    /// a filename including the extension needs to be provided,
    /// otherwise the file won't be found.
```

Ultimately I think we need a lot more documentation on the way file names are resolved and run, including caveats, but that's perhaps a bigger job than this PR. And until we do I'm nervous about actively encouraging people to use `.bat` files in `Command::new` because it was an accidental feature that we've just tried not to break too badly.

**Review by ChrisDenton — COMMENTED** [review]
For sure! As per my comment above I am generally in favour of this but have one concern.

**Comment by sanchopanca** [comment]
@ChrisDenton thanks for the review. I have incorporated you suggestions.

**Comment by ChrisDenton** [comment]
Thanks!

@bors r+ rollup
